hpx/parallel/segmented_algorithms/transform.hpp

```diff
@@ -0,0 +1,876 @@
+                                    beg1, end1, beg2, end2, ldest, f, proj1, proj2)
+                            );
+                        }
+                    }
+                }
+
+                // handle the beginning of the last partition
+                beg1 = traits1::begin(sit1);
+                beg2 = traits2::begin(sit2);
+                end1 = traits1::end(sit1);
+                end2 = traits2::end(sit2);
+                if (beg1 != end1 && beg2 != end2)
+                {
+                    segments.push_back(
+                        dispatch_async(traits1::get_id(sit1),
+                            algo, policy, forced_seq(),
+                            beg1, end1, beg2, end2, ldest, f, proj1, proj2)
+                    );
+                }
+            }
+
+            return result::get((
+                dataflow(
+                    [=](std::vector<hpx::future<hpx::util::tuple<
+                        local_iterator_type1, local_iterator_type2,
+                        local_iterator_type3> > > && r)
+                         -> hpx::util::tuple<InIter1, InIter2, OutIter>
+                    {
+                        // handle any remote exceptions, will throw on error
+                        std::list<std::exception_ptr> errors;
+                        parallel::util::detail::handle_remote_exceptions<
+                            ExPolicy
+                        >::call(r, errors);
+                        auto rl = r.back().get();
+                        auto olast1 = traits1::compose(send1, hpx::util::get<0>(rl));
+                        auto olast2 = traits2::compose(send2, hpx::util::get<1>(rl));
+                        auto odest = traits3::compose(sdest, hpx::util::get<2>(rl));
+                        return hpx::util::make_tuple(olast1, olast2, odest);
+                    },
+                    std::move(segments))));
+        }
+
+        ///////////////////////////////////////////////////////////////////////
+        // segmented implementation
+        template <
+            typename ExPolicy, typename InIter1, typename InIter2,
+            typename OutIter, typename F, typename Proj1, typename Proj2>
+        typename util::detail::algorithm_result<
+            ExPolicy,
+            hpx::util::tagged_tuple<
+                tag::in1(InIter1), tag::in2(InIter2), tag::out(OutIter)
+            >
+        >::type
+        transform_(ExPolicy && policy,
+            InIter1 first1, InIter1 last1, InIter2 first2, InIter2 last2,
+            OutIter dest, F && f, Proj1 && proj1,
+            Proj2 && proj2, std::true_type)
+        {
+            typedef parallel::execution::is_sequenced_execution_policy<
+                    ExPolicy
+                > is_seq;
+            typedef util::detail::algorithm_result<ExPolicy,
+            hpx::util::tagged_tuple<
+                tag::in1(InIter1), tag::in2(InIter2), tag::out(OutIter)>
+            > result;
```

Comment:
The indentation is not consistent here, e.g. why the indentation style for `tagged_tuple` is different in lines 820 and 833?